include/seqan/graph_algorithms/depth_first_search.h

```diff
@@ -121,28 +123,72 @@ void depthFirstSearch(TPredecessorMap & predecessor,
     typedef typename Iterator<TGraph, VertexIterator>::Type TVertexIterator;
     typedef typename VertexDescriptor<TGraph>::Type TVertexDescriptor;
     typedef typename Value<TPredecessorMap>::Type TPredVal;
+    typedef typename Iterator<Graph<TSpec>, AdjacencyIterator>::Type TAdjacencyIterator;
+    enum Task {EXPLORE, FINISH};
```

Comment:
Please make this a typed enum using class keyword and probably use ```uint8_t``` as underlying type.
Maybe also separate this enum definition with a new line from the typedefs above.